SharedSources/MediaLibraryModel/ArtistModel.swift

```diff
@@ -37,6 +37,27 @@ class ArtistModel: MLBaseModel {
     func delete(_ items: [VLCMLObject]) {
         preconditionFailure("ArtistModel: Cannot delete artist")
     }
+
+    func createPlaylist(_ name: String, _ fileIndexes: Set<IndexPath>? = nil) {
+        let playlist = medialibrary.createPlaylist(with: name)
+
+        guard let fileIndexes = fileIndexes else {
+            return
+        }
+
+        for index in fileIndexes  where index.row < files.count {
+            // Get all tracks from a VLCMLArtist
+            guard let tracks = files[index.row].tracks(with: .default, desc: false) else {
+                assertionFailure("ArtistModel: createPlaylist: Fail to retreive tracks.")
+                return
+            }
```

Comment:
I would suggest having a tracks convenience method on Medialibrary here. I believe using the sorting method to access the tracks here is unintuitive. But mostly because every sorting call to this method triggers calls to the db which will be very slow for many tracks and it can potentially get out of sync between sorting calls if a file is added while creating the playlist.

Reply:
I would prefer merging this and adapting it once the changes are made on the medialibrary side if that is okay with you.

Reply:
okay under the condition that you/we work next on it so that this gets addressed soon. We run into the same issue with the CollectionViewcontroller for genre, artist. We can't just use the sortcall everywhere where we use genres and tracks and have multiple cached versions of the sorting result :/ That basically screams for problems

Reply:
Actually, if we look into the C++ code, it is basically the same call, there is no "sort" call. Indeed, there is only one way of getting the value and depending on the parameter given it will change.

The integration of the tracks in album is actually incorrect since it only checks if it exists. So in the case where the album is updated, the tracks are not.

I asked Hugo about this and he recommends that the wrapper should directly give the value from the medialib without caching anything and if we want to optimize it, we should do it on the application side.